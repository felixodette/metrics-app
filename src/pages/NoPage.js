import React from 'react';

const NoPage = () => (
  <div>Page not found!</div>
);

export default NoPage;
