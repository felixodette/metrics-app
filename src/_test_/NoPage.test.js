import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import NoPage from '../pages/NoPage';

describe('Create header', () => {
  it('Render without errors', () => {
    const tree = renderer
      .create(
        <Router>
          <NoPage />
        </Router>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the elements', () => {
    render(
      <Router>
        <NoPage />
      </Router>,
    );
    const element = screen.getByText('Page not found!');
    expect(element).toBeInTheDocument();
  });
});
