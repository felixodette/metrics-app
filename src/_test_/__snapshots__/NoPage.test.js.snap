// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create header Render without errors 1`] = `
<div>
  Page not found!
</div>
`;
